resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -101,55 +152,82 @@ class TimeZone {
 
         $currentState = $this.Get()
 
-        if ($currentState.TimeZone -ne $this.TimeZone) {
-            Set-TimeZone -Id (Get-ValidTimeZone -TimeZone $this.TimeZone)
+        if ($currentState.Id -ne $this.Id) {
+            Set-TimeZone -Id (Get-ValidTimeZone -TimeZone $this.Id)
+        }
+
+        if (($null -ne $this.SetTimeZoneAutomatically) -and ($this.SetTimeZoneAutomatically -ne $currentState.SetTimeZoneAutomatically)) {
+            $desiredState = $this.SetTimeZoneAutomatically ? 3 : 4
+
+            if (Test-LocationSettingPermission) {
+                Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([TimeZone]::SetTimeZoneAutomaticallyProperty) -Value $desiredState
+            }
         }
 
-        if ($currentState.SetTimeZoneAutomatically -ne $this.SetTimeZoneAutomatically) {
+        if (($null -ne $this.SetTimeAutomatically) -and ($this.SetTimeAutomatically -ne $currentState.SetTimeAutomatically)) {
             $desiredState = $this.SetTimeAutomatically ? [TimeZone]::NtpEnabled : [TimeZone]::NtpDisabled
 
-            Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([TimeZone]::SetTimeZoneAutomaticallyProperty) -Value $desiredState
+            Set-ItemProperty -Path $global:w32TimePath -Name ([TimeZone]::SetTimeAutomaticallyProperty) -Value $desiredState
         }
 
-        if ($currentState.AdjustForDaylightSaving -ne $this.AdjustForDaylightSaving) {
-            $desiredState = $this.AdjustForDaylightSaving ? 0 : 1
+        if ($currentState::SupportsDaylightSavingProperty -and ($null -ne $this.AdjustForDaylightSaving)) {
+            $desiredState = @{
+                Id     = $this.Id
+                Enable = $this.AdjustForDaylightSaving
+            }
 
-            Set-ItemProperty -Path $global:timeZoneInformationPath -Name ([TimeZone]::AdjustForDaylightSavingProperty) -Value $desiredState
+            if ($this.AdjustForDaylightSaving -ne $currentState.AdjustForDaylightSaving) {
+                # Falling back on tzutil because the amount of registry keys to modify is too high, plus difficult to determine which one to modify
+                Set-DaylightSavingTime @desiredState
+            }
```

Comment:
What happens if you try to enable daylight saving using tzutil on an id that doesn't support daylight saving

Reply:
The function `Set-DaylightSavingTime` checks the LASTEXITCODE. IF it not equals 0, it fails and throws an error.